### DISCUSSION TITLE
When writing data, the latest data cannot be queried

### DISCUSSION THREAD

**Discussion by cyshooter** [duckdb/duckdb#7657] [Q&A] [answered]
hello
    i'm using java appender insert data to duckdb, But when using python to query data, only a specific data set is returned and there is no change.
OS: windows 11
duckdb version:0.8.0
jdk:8
python:3.11
java code:

                                Properties ro_prop = new Properties();
				ro_prop.setProperty("duckdb.read_only", "true");
				Class.forName("org.duckdb.DuckDBDriver");
				conn = DriverManager.getConnection("jdbc:duckdb:C://software//duckdb//data//shinetest.duckdb",ro_prop);
                                appender =((DuckDBConnection) conn).createAppender("main", "vibdata");
                               StringBuilder stb = new StringBuilder(50);			
			      for(int i=0;i<row;i++)
			      {				
				stb.append(dateFormat.format(date) + String.format("%03d", microsecond % 1000));
				appender.beginRow();
				appender.append(stb.toString());
				for(int j=0;j<width;j++)
				{
					appender.append(vib.getData()[i][j]);					
				}
				appender.endRow();
				stb.setLength(0);
				microsecond+=975;
				incremic+=975;
				date=(date/1000)*1000+(incremic/1000);
			    }         
                            appender.flush();

python code:

import duckdb
con=duckdb.connect(database='C://software//duckdb//data//shinetest.duckdb',read_only=True)
con.execute("SELECT timestamp,ch1 FROM vibdata order by timestamp desc  limit 20")
print(con.fetchall())
con.close

then run python code,always return the same dataset, but java program is still inserting data.

**Comment by Mytherin** [comment, marked as answer]
Having one process write data to a database while another process reads from it is not supported, so this is not expected to work.

I am guessing the Java program is not actually appending data as otherwise an error should have been thrown, as you cannot append to a persistent database in read-only mode.

**Reply by cyshooter** [reply]
thanks a lot
